test/e2e/provisioning/all_e2e_test.go

```diff
@@ -0,0 +1,86 @@
+// +build e2e
+
+package provisioning
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+const (
+	do_manifest  = "./testdata/machine-digitalocean.yaml"
```

Comment:
I'd prefer to keep them separate. 

The ones under `examples/` are for the operators that wish to use the machine-controller. Thus, most of the fields are filled out for them. They would have to only provide credentials.

The ones under `testdata/` are used by `e2e` test and are prepared in a way that suits theirs needs.

Reply:
Well what bothers me a little about keeping them seperate is that it means our examples are completely untested and they are the first thing a new user will see, so that potentially results in a poor experience ;)

Do you think it would be feasible to use the example, considering we currently replace all fields that need to be configured test-specific anyways?